src/org/labkey/test/components/ui/grids/CustomizeGridDialog.java

```diff
@@ -0,0 +1,491 @@
+        else
+        {
+            elementCache().checkbox.uncheck();
+        }
+
+        return this;
+    }
+
+    /**
+     * Is the 'Show all system and user-defined fields' checkbox checked?
+     *
+     * @return True if checked false otherwise.
+     */
+    public boolean isShowAllChecked()
+    {
+        return elementCache().checkbox.isChecked();
+    }
+
+    /**
+     * Get the list of visible fields from the 'Available Fields' panel. Children of an expanded filed only have the name
+     * in the list, this des not include the path.
+     *
+     * @return The list of field names.
+     */
+    public List<String> getAvailableFields()
+    {
+        List<WebElement> listItemElements = elementCache().getListItemNameElements(elementCache().availableFieldsPanel());
+        return listItemElements.stream().map(WebElement::getText).collect(Collectors.toList());
+    }
+
+    /**
+     * Add a top level field to the 'Shown in Grid' list.
+     *
+     * @param fieldName The name of the field.
+     * @return This dialog.
+     */
+    public CustomizeGridDialog addAvailableFieldToGrid(String fieldName)
+    {
+        WebElement listItem = elementCache().getListItemElementByFieldKey(elementCache().availableFieldsPanel(), fieldName);
+
+        Assert.assertTrue(String.format(FIELD_NOT_AVAILABLE, fieldName),
+                listItem.isDisplayed());
+
+        WebElement addIcon = Locator.tagWithClass("div", "view-field__action")
+                .withChild(Locator.tagWithClass("i", "fa-plus"))
+                        .findElement(listItem);
+
+        addIcon.click();
+
+        return this;
+    }
+
+    /**
+     * Add a nested field to the 'Shown in Grid' list. You must provide the list of parent fields.
+     *
+     * @param parentFields The list of parent fields to first expand.
+     * @param fieldName The field to add.
+     * @return This dialog.
+     */
+    public CustomizeGridDialog addAvailableFieldToGrid(List<String> parentFields, String fieldName)
+    {
+        StringBuilder fieldKey = new StringBuilder();
+
+        for(String field : parentFields)
+        {
+            fieldKey.append(field.replace(" ", ""));
+            expandOrCollapseByFieldKey(fieldKey.toString(), true);
+            fieldKey.append("/");
+        }
+
+        fieldKey.append(fieldName);
+
+        return addAvailableFieldToGrid(fieldKey.toString());
+    }
+
+    /**
+     * Expand a top level field in the 'Available Fields' panel.
+     *
+     * @param fieldName The field to expand.
+     * @return This dialog.
+     */
+    public CustomizeGridDialog expandAvailableField(String fieldName)
+    {
+        expandOrCollapseByFieldKey(fieldName.replace(" ",""), true);
+        return this;
+    }
+
+    /**
+     * Expand a hierarchy of fields.
+     *
+     * @param parentFields The list of fields to expand.
+     * @return This dialog.
+     */
+    public CustomizeGridDialog expandAvailableFieldsTo(List<String> parentFields)
```

Comment:
Combine both `expandAvailableField` methods into one varargs method:
```suggestion
    public CustomizeGridDialog expandAvailableFieldsTo(String... fieldKeyParts)
```

Reply:
Done.